packages/world/src/SystemCall.sol

```diff
@@ -74,6 +74,57 @@ library SystemCall {
       });
   }
 
+  /**
+   * @notice Makes a staticcall to a system identified by its Resource ID while ensuring necessary access controls.
+   * @dev This function does not revert if the system staticcall fails. Instead, it returns a success flag.
+   * @param caller The address initiating the system staticcall.
+   * @param systemId The unique Resource ID of the system being called.
+   * @param callData The calldata to be executed in the system.
+   * @return success A flag indicating whether the system staticcall was successful.
+   * @return data The return data from the system staticcall.
+   */
+  function staticcall(
+    address caller,
+    ResourceId systemId,
+    bytes memory callData
+  ) internal view returns (bool success, bytes memory data) {
+    // Load the system data
+    (address systemAddress, bool publicAccess) = Systems._get(systemId);
+
+    // Check if the system exists
+    if (systemAddress == address(0)) revert IWorldErrors.World_ResourceNotFound(systemId, systemId.toString());
+
+    // Staticcalls are not supported for root systems
```

Comment:
Should we add some context for why they're not supported for root systems in this comment? Might not be obvious on first sight